primer/src/Primer/API.hs

```diff
@@ -1109,23 +1109,23 @@ evalFull' ::
   GVarName ->
   PrimerM m EvalFullResp
 evalFull' = curry3 $ logAPI (noError EvalFull') $ \(sid, lim, d) ->
-  noErr <$> liftEditAppM (q lim d) sid
+  noErr <$> liftQueryAppM (q lim d) sid
   where
     q ::
       Maybe TerminationBound ->
       GVarName ->
-      EditAppM (PureLog (WithSeverity l)) Void EvalFullResp
+      QueryAppM (PureLog (WithSeverity l)) Void EvalFullResp
     q lim d = do
-      e <- DSL.gvar d
-      a <- get
+      -- We don't care about uniqueness of this ID, and we do not want to
+      -- disturb any FreshID state, since that could break undo/redo.
```

Comment:
It's not _immediately_ obvious why this is safe. I guess because the first eval step will always be to inline the definition, thus removing the ID? Anyway, this is probably worth a comment.